setup.py

```diff
@@ -10,21 +10,71 @@
 
 import os
 import re
+import runpy
 import subprocess
 
 from setuptools import setup, find_packages
 
+
+# Version information; update this by hand when making a new bugfix or feature
+# release. The actual package version is autogenerated from this information
+# together with information from the version control system, and then injected
+# into the package source.
 MAJOR = 7
 MINOR = 2
 MICRO = 0
-
+PRERELEASE = ""
 IS_RELEASED = False
 
-VERSION = "%d.%d.%d" % (MAJOR, MINOR, MICRO)
+# If this file is part of a Git export (for example created with "git archive",
+# or downloaded from GitHub), ARCHIVE_COMMIT_HASH gives the full hash of the
+# commit that was exported.
+ARCHIVE_COMMIT_HASH = "$Format:%H$"
+
+# Templates for version strings.
+RELEASED_VERSION = "{major}.{minor}.{micro}{prerelease}"
+UNRELEASED_VERSION = "{major}.{minor}.{micro}{prerelease}.dev{dev}"
```

Comment:
just to confirm, here's what i think the process needs to be based on this PR

- `PRERELEASE = ''` set to empty string up until a release
- before making a prerelease, we flip `IS_RELEASED=True` and set `PRERELEASE='rc0'`
- when we are reading to make the release, we set `PRERELEASE=''` set back to empty string.

Is that correct?

Reply:
This is also my understanding. That said, this is also taken from Traits as I am trying to make things more consistent.
(For my own personal projects, I compute the version differently 😂  so there isn't a "IS_RELEASED" boolean.)

Reply:
the exact same question I had is what Corran also asked in https://github.com/enthought/traits/pull/846/files#r372846588 - to which Mark responsed in the thread.

It would be useful to document this somewhere though (either in the setup.py or in the checklist) instead of in this thread/in our brains.

Reply:
My thought is that this will take another release to exercise the full release process including the release candidate, and while we do that, we can document the steps like we did in #714, with references to PRs making these steps. Eventually when the same steps have been demonstrated to work a second time, we can write them in a separate document if we like. But repositories are going to have slight variations (like TraitsUI having to update ci-src-requirements.txt while Pyface does not), so the issue from the previous release is always going to the one I would look for when I make a release.

Reply:
See #714 where I have updated the steps as I went, documenting what happened for 7.1 and the steps that had not happened because of #744.

Reply:
> like TraitsUI having to update ci-src-requirements.txt while Pyface does not

My bad of jumping into conclusion before verifying. Turns out pyface installs traitsui master too, so the ci-src-requirements.txt in pyface needs to be updated for the maintenance branch too.